src/module-elasticsuite-swatches/Block/Navigation/Renderer/Swatches/RenderLayered.php

```diff
@@ -67,7 +67,9 @@ public function getSwatchData(): array
         }
 
         foreach (array_merge(array_flip($sortingArr), $attributeOptions) as $item) {
-            $attributeOptionsSort[$item['id']] = $item;
+            if (isset($item) && is_array($item)) {
```

Comment:
Could you remove the `isset($item)` test ? It's not an accurate test to know if a key exists in an array and besides, we know that $item exists, since it is the byproduct of the loop (ie the current item).
I would go for `is_array($item) && array_key_exists('id', $item)`.

The original issue due to the fact that the label is " XXX" and the indexed label is "XXX" should be taken care of, but considering we have another issue in the area (#2029), we'll maybe look at it then.